packages/engine/errors/8016.md

```diff
@@ -0,0 +1,6 @@
+---
+original: "Type assertion expressions can only be used in TypeScript files."
+excerpt: "This isn't a Typescript file"
+---
+
+You're trying to use Typescript but we aren't inside of a .ts file.
```

Comment:
TS can be used in a `.d.ts`, `.ts`, `.tsx` file, or `.mts` file - I'd say leave out the reference to `.ts` to avoid confusion.